ui: Automate formatting standards

vladlos opened the pull request:
Added packages `Eslint and prettier` for checking and fixing `lint` rules.
Added packages husky and lint-staged for make pre-commit check on lint and tests, if something is failed do not commit

Importing config for `Eslint` from @cockroachlabs/eslint-config.
Fixed all eslint and tslint rules.

Added additional rules into `.eslintrc` for avoiding some mistakes.

Release justification: non-production code changes

Release note (ui): database loading state design updates

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47339)
<!-- Reviewable:end -->

dhartunian commented:
@elkmaster does this need a `yarn-vendored` update as well? I don't think I see one included.

vladlos committed:
ui: Automate formatting standards

Added packages `Eslint and prettier` for checking and fixing `lint` rules.
Added packages husky and lint-staged for make pre-commit check on lint and tests, if something is failed do not commit

Importing config for `Eslint` from @cockroachlabs/eslint-config.
Fixed all eslint and tslint rules.

Added additional rules into `.eslintrc` for avoiding some mistakes.

Release note (ui): database loading state design updates
